azure-auth/azure_auth/auth/core.py

```diff
@@ -0,0 +1,68 @@
+from urllib.parse import parse_qs, urlparse
+import msal
+import reflex as rx
+from typing import Dict
+
+client_id: str = "0df2a88e-fddb-4cc2-b3e0-f475f162b373"
+client_secret: str = ""
+tenant_id: str = "f2c9cbbe-006b-46b8-9ad0-d877d8446d6d"
+redirect_uri: str = "http://localhost:3000/callback"
```

Comment:
this is problematic to hardcode such a url like this, probably better to use `router` data to figure out where the frontend is actually being accessed from.

in an event handler, use `f"{self.router.page.host}/callback"`

https://reflex.dev/docs/utility-methods/router-attributes/